code/modules/reagents/chemistry/reagents/unique/eigenstasium.dm

```diff
@@ -0,0 +1,111 @@
+//During the process you get really hungry, then some of your items slowly start teleport around you,
+//then alternative versions of yourself are brought in from a different universe and they yell at you.
+//and finally you yourself get teleported to an alternative universe, and character your playing is replaced with said alternative
+
+/datum/reagent/eigenstate
+	name = "Eigenstasium"
+	description = "A strange mixture formed from a controlled reaction of bluespace with plasma, that causes localised eigenstate fluxuations within the patient"
+	taste_description = "wiggly cosmic dust."
+	color = "#5020F4"
+	overdose_threshold = 15
+	metabolization_rate = 1 * REAGENTS_METABOLISM
+	ph = 3.7
+	impure_chem = /datum/reagent/impurity/eigenswap
+	inverse_chem = null
+	failed_chem = /datum/reagent/bluespace //crashes out
+	chemical_flags = REAGENT_DEAD_PROCESS //So if you die with it in your body, you still get teleported back to the location as a corpse
+	data = list("location_created" = null, "creator" = null)//So we retain the target location and creator between reagent instances
+	///The creation point assigned during the reaction
+	var/turf/location_created
+	///The mob that initial teleport is linked to
+	var/mob/living/carbon/creator
+	///The return point indicator
+	var/obj/effect/overlay/holo_pad_hologram/eigenstate
+	///The point you're returning to after the reagent is removed
+	var/turf/open/location_return = null
+
+/datum/reagent/eigenstate/on_new(list/data)
+	location_created = data["location_created"]
+	creator = data["creator"]
+
+/datum/reagent/eigenstate/expose_mob(mob/living/living_mob, methods, reac_volume, show_message, touch_protection)
+	. = ..()
+	if(!(methods & INGEST))
+		return
+	if(creation_purity > 0.9 && location_created) //Teleports you home if it's pure enough
+		do_sparks(5,FALSE,living_mob)
+		do_teleport(living_mob, location_created, 0, asoundin = 'sound/effects/phasein.ogg')
+		do_sparks(5,FALSE,living_mob)
+
+//Main functions
+/datum/reagent/eigenstate/on_mob_add(mob/living/living_mob, amount)
+	//make hologram at return point
+	eigenstate = new (living_mob.loc)
+	eigenstate.appearance = creator ? creator.appearance : living_mob.appearance //Blood lets you set your eigenstate icon
+	eigenstate.alpha = 170
+	eigenstate.add_atom_colour("#77abff", FIXED_COLOUR_PRIORITY)
+	eigenstate.mouse_opacity = MOUSE_OPACITY_TRANSPARENT//So you can't click on it.
+	eigenstate.layer = FLY_LAYER//Above all the other objects/mobs. Or the vast majority of them.
+	eigenstate.anchored = 1//So space wind cannot drag it.
+	eigenstate.name = "[creator ? creator.appearance : living_mob.appearance]'s' eigenstate"//If someone decides to right click.
+	eigenstate.set_light(2)	//hologram lighting
+
+	location_return = get_turf(living_mob)	//sets up return point
+	to_chat(living_mob, "<span class='userdanger'>You feel your wavefunction split!</span>")
+
+	. = ..()
```

Comment:
Any reason why not to call parent first?
Same for `on_mob_life()`, `on_mob_delete()` `overdose_start()` and `overdose_process()`.

Reply:
Reagents removes volumes and increments cycle in the parent call - so I generally call if afterwards, but I can move it before if you believe that to be the best. I just don't know how del_reagent() will work with on_mob_life() when called by remove_reagent()

Reply:
That's fine. If it's so, make it `return ..()` instead of `. = ..()`, when it's at the end of the proc.
Just for clarity, since `. = ..()` implies there's more code after, instead of it being the end of the proc.

Reply:
Done